packages/openapi-v3/src/json-to-schema.ts

```diff
@@ -75,7 +75,9 @@ export function jsonToSchemaObject(json: JsonSchema): SchemaObject | SchemaRef {
         break;
       }
       case 'additionalProperties': {
-        if (typeof json.additionalProperties !== 'boolean') {
+        if (typeof json.additionalProperties === 'boolean') {
```

Comment:
IIUC `additionalProperties` indicates that if the additional properties are allowed, it also converts JSON to object. 

Just out of curiosity, why can't we use `strict` field to decide if additional properties are allowed and `additionalProperties` to convert types when `strict` is true?

Reply:
I see, thanks for the clarification!